### PR TITLE
Remove legacy Entity Linker component

### PR DESCRIPTION
**Pull request by polm**
## Description

The Entity Linker was unusual in that the component (rather than the model) had to be modified due to issues, creating a legacy version. This PR removes the legacy version and one test that covered it.

### Types of change

Cleanup.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — COMMENTED** [review]
So that would mean that people won't be able to use the v1 anymore as soon as they update to spaCy v4, right? 

How much effort is it to update a config file with v1 to v2?

**Comment by polm** [comment]
It's correct this means that v1 won't work with spaCy v4 any longer. 

Updating the config shouldn't require any changes besides changing the version - only one option was added, it has a reasonable default, and none of the other options changed. Changes from v1 to v2 were primarily motivated by a bugfix related to listeners, and though it ended up changing the internals a lot, it shouldn't affect the accuracy or anything. However, if I remember correctly, because the structure of the Thinc models changed, it does require retraining.

For reference, v2 was added and v1 was made legacy in #9669.

**Comment by polm** [comment]
Here's a thread where I confirmed the need to retrain. https://github.com/explosion/spaCy/discussions/10887#discussioncomment-2885821

**Comment by polm** [comment]
I reviewed this again and it should be good to go.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
       matrix:
         # We're only running one platform per Python version to speed up builds
         Python36Linux:
-          imageName: "ubuntu-latest"
+          imageName: "ubuntu-20.04"
```
Comment (rmitsch) [inline azure-pipelines.yml:44]: Out of curiosity, why this change?

**Review by rmitsch — APPROVED** [review]
I think this it's alright to remove this for v4 (and that retraining is an acceptable compromise). Approving, but let's double-check with @svlandeg before merging.

File: azure-pipelines.yml (hunk 1 above)
Comment (polm) [reply azure-pipelines.yml:44]: See #11928 - `ubuntu-latest` changed a little earlier than we expected, and it drops 3.6.

Comment (svlandeg) [inline azure-pipelines.yml:44]: I just merged the updates from `master` to `v4` which contain this commit: https://github.com/explosion/spaCy/commit/8267aa1b65783ea29e42f470eaf7553ff7508206
So it's probably best if you merge in the latest `v4` again so we don't end up with merge conflicts :-)

**Review by svlandeg — CHANGES_REQUESTED** [review]
As discussed internally, can we first explore the option to move this code to `spacy-legacy` instead?

**Comment by polm** [comment]
I'll mark this as draft while we figure out how to handle this.

**Comment by polm** [comment]
Since we'll continue to support this in `spacy-legacy`, let me make a new PR rather than reverting the only change here.
